source/server/BUILD

```diff
@@ -257,9 +257,28 @@ envoy_cc_library(
 )
 
 envoy_cc_library(
-    name = "listener_manager_lib",
-    srcs = ["listener_manager_impl.cc"],
-    hdrs = ["listener_manager_impl.h"],
+    name = "listener_lib",
+    srcs = [
+        "listener_impl.cc",
+    ],
+    hdrs = [
+        "listener_impl.h",
+    ],
+    deps = [
+        ":listener_manager_impl",
+        "//source/common/network:connection_balancer_lib",
```

Comment:
I think there are a number of direct dependencies not reflected here. I know this is a pain, but I think it's worth looking through the includes in listenter_impl.h/.cc and copying over the corresponding dependencies from the the listener_manager_lib target.

Reply:
May I ask why?
The context is listener_manager_impl and listener_impl are coupled. They depends on implementation instead of interface.
At this moment, and in short future, `Listener_impl` cannot be built without a `listener_manager_impl`
What I did is to move deps from listener_manager_impl to here: connection_balancer_lib, and remove obsolete deps: see the red diff in listener_manager_impl

Reply:
Because they are direct dependencies of listener_impl.h/.cc and as listener_manager_impl.h/.cc is modified and its dependencies modified this target will need to be updated.

Reply:
Updated. @zuercher Could you take a look?